Add licensecc port

hanjingo opened the pull request:
## Description
This PR adds a new port for licensecc, a cross-platform C++ license management library with hardware fingerprinting and trial support.

## Checklist
- [x] Changes comply with the [maintainer guide](https://github.com/Microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/Microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches are formatted with `git format-patch` and validated with `git am`
- [x] License is correctly identified in vcpkg.json
- [x] Port builds successfully on tested platforms

## Testing
- [x] Windows x64 - Visual Studio 2019
- [ ] Linux x64 (needs CI testing)
- [ ] macOS x64 (needs CI testing)

## Additional Notes
- This port includes patches for Boost.Filesystem compatibility
- Project initialization is bypassed as it requires license generation tools not needed for library usage
- Library follows BSD-3-Clause license terms

hanjingo committed:
Add licensecc port

Adds support for licensecc, a cross-platform C++ license management library.

- Version: 2.0.0
- License: BSD-3-Clause
- Features: Hardware fingerprinting, trial support
- Tested on: Windows x64

hanjingo committed:
[licensecc] Add new port

- Add licensecc 2.0.0, a cross-platform C++ license management library
- Features hardware fingerprinting and trial support
- Provides static library with OpenSSL backend
- Includes CMake config files for easy integration
- Fixes for Boost.Filesystem compatibility and header includes

BillyONeal commented on ports/licensecc/vcpkg.json:
```diff
@@ -0,0 +1,19 @@
+  "license": "BSD-3-Clause",
+  "supports": "!uwp",
+  "dependencies": [
+    "openssl",
```
```
[2/2] "/vcpkg/downloads/tools/cmake-3.30.1-linux/cmake-3.30.1-linux-x86_64/bin/cmake" -E chdir ".." "/vcpkg/downloads/tools/cmake-3.30.1-linux/cmake-3.30.1-linux-x86_64/bin/cmake" "/mnt/vcpkg-ci/b/licensecc/src/v2.0.0-d4b7d35fcf.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=/mnt/vcpkg-ci/p/licensecc_x64-linux" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DBUILD_SHARED_LIBS=OFF" "-DBUILD_TESTING=OFF" "-DLCC_LOCATION=OFF" "-DCMAKE_DISABLE_FIND_PACKAGE_Git=ON" "-DCMAKE_MAKE_PROGRAM=/vcpkg/downloads/tools/ninja/1.12.1-linux/ninja" "-DCMAKE_SYSTEM_NAME=Linux" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/vcpkg/scripts/toolchains/linux.cmake" "-DVCPKG_TARGET_TRIPLET=x64-linux" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/vcpkg" "-D_VCPKG_INSTALLED_DIR=/mnt/vcpkg-ci/installed" "-DVCPKG_MANIFEST_INSTALL=OFF"
FAILED: ../CMakeCache.txt 
"/vcpkg/downloads/tools/cmake-3.30.1-linux/cmake-3.30.1-linux-x86_64/bin/cmake" -E chdir ".." "/vcpkg/downloads/tools/cmake-3.30.1-linux/cmake-3.30.1-linux-x86_64/bin/cmake" "/mnt/vcpkg-ci/b/licensecc/src/v2.0.0-d4b7d35fcf.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=/mnt/vcpkg-ci/p/licensecc_x64-linux" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DBUILD_SHARED_LIBS=OFF" "-DBUILD_TESTING=OFF" "-DLCC_LOCATION=OFF" "-DCMAKE_DISABLE_FIND_PACKAGE_Git=ON" "-DCMAKE_MAKE_PROGRAM=/vcpkg/downloads/tools/ninja/1.12.1-linux/ninja" "-DCMAKE_SYSTEM_NAME=Linux" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/vcpkg/scripts/toolchains/linux.cmake" "-DVCPKG_TARGET_TRIPLET=x64-linux" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/vcpkg" "-D_VCPKG_INSTALLED_DIR=/mnt/vcpkg-ci/installed" "-DVCPKG_MANIFEST_INSTALL=OFF"
-- You should define a variable LCC_PROJECT_NAME containing the name of the software you want to add a license to.A mock project named DEFAULT has been added for you.
-- The CXX compiler identification is GNU 11.4.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2")
-- Could NOT find lccgen (missing: lccgen_DIR)
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found openssl version 3.5.2 - generator
CMake Warning (dev) at /vcpkg/scripts/buildsystems/vcpkg.cmake:896 (_find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

Call Stack (most recent call first):
  extern/license-generator/CMakeLists.txt:102 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at /vcpkg/downloads/tools/cmake-3.30.1-linux/cmake-3.30.1-linux-x86_64/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find Boost (missing: Boost_INCLUDE_DIR date_time filesystem
  program_options system unit_test_framework)
Call Stack (most recent call first):
  /vcpkg/downloads/tools/cmake-3.30.1-linux/cmake-3.30.1-linux-x86_64/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  /vcpkg/downloads/tools/cmake-3.30.1-linux/cmake-3.30.1-linux-x86_64/share/cmake-3.30/Modules/FindBoost.cmake:2409 (find_package_handle_standard_args)
  /vcpkg/scripts/buildsystems/vcpkg.cmake:896 (_find_package)
  extern/license-generator/CMakeLists.txt:102 (find_package)
```

Looks like you are missing a lot of boost dependencies.

hanjingo committed:
Merge branch 'microsoft:master' into add-licensecc-port

hanjingo commented:
> @hanjingo please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement



> @hanjingo please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree [company="{none}"]

hanjingo committed:
[licensecc] Update to 2.0.0#3

- Restore CMake config files for proper target integration
- Remove unnecessary Boost filesystem normalization patch
- Improve CMake configuration with proper OpenSSL dependency linking
- Enable both debug and release library configurations
- Ensure compatibility with vcpkg's CMake integration pattern

hanjingo committed:
Merge branch 'add-licensecc-port' of github.com:hanjingo/vcpkg into add-licensecc-port

hanjingo committed:
[licensecc] Add new port for cross-platform license management

- Add licensecc 2.0.0 port with support for x64-windows and x86-windows
- Includes hardware fingerprinting and license validation features
- Fixed x86-windows build by disabling Boost-dependent components
- Provides CMake targets for easy integration
- Supports OpenSSL-based cryptographic operations

Tested on:
- x64-windows: builds successfully
- x86-windows: builds successfully

Resolves cross-platform license management requirements.

BillyONeal commented:
Since there still seems to be a lot of build failures I've marked this as draft for now; please feel free to mark "Ready for review" when you're ready for us to look. Thanks for the new port!
